### PR TITLE
Refactor status icons take 2, cyborgs don't see criminal status

### PR DESCRIPTION
**Pull request by Krunklehorn**
## About the PR

Cleans up and refactors the basic status icon systems.

Prevents cyborgs from seeing criminal record status.

Continuation of #26133 without grouping under one system.

## Why / Balance

Put briefly, cyborgs aren't part of Security and don't need to see criminal status icons. I argue that giving them this information encourages poor cyborg play.

https://discord.com/channels/310555209753690112/1218078327504506960

## Technical details

`ShowSecurityIcons` was split into `ShowJobIcons`, `ShowMindShieldIcons` and `ShowCriminalRecordIcons`.

Prototype indexes for `ShowHungerIconsSystem` and `ShowThirstIconsSystem` are now cached in `HungerSystem` and `ThirstSystem` respectively.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Krunk
- remove: Cyborgs no longer see criminal status icons.

### CODE REVIEW

**Comment by Krunklehorn** [comment]
If #26004 is to be merged I'll add the clause to `ShowJobIcons`.

**Comment by Hmeister-real** [comment]
it would be good if you couldn't see the icons at all if people are in darkness, or inside a container

**Comment by Krunklehorn** [comment]
> it would be good if you couldn't see the icons at all if people are in darkness

Beyond the scope of this PR.

**Comment by superjj18** [comment]
how does Borgs having access to the legal standing of a crewmate encourage poor cyborg play?  Currently the worst they can do is snitch on a wanted crewmate, if they do beyond that they’ve broken rules.  I just don’t see the value in it, and I think Borgs should at very least be informed as a trade off for being law-bound

**Comment by Krunklehorn** [comment]
> how does Borgs having access to the legal standing of a crewmate encourage poor cyborg play? Currently the worst they can do is snitch on a wanted crewmate, if they do beyond that they’ve broken rules. I just don’t see the value in it, and I think Borgs should at very least be informed as a trade off for being law-bound

This was covered in the feedback thread multiple times.

In short, cyborgs aren't Sec and the players controlling them don't need Sec-relevant information clouding their decision process.

File: Content.Client/Overlays/ShowHungerIconsSystem.cs [hunk 1]
```diff
@@ -17,42 +16,12 @@ public override void Initialize()
...
-                if (_prototypeMan.TryIndex<StatusIconPrototype>("HungerIconStarving", out var starving))
-                {
-                    result.Add(starving);
-                }
-                break;
-        }
-
-        return result;
+        if (_hunger.TryGetStatusIconPrototype(component, out var iconPrototype))
+            ev.StatusIcons.Add(iconPrototype!);
```
Comment (metalgearsloth) [inline Content.Client/Overlays/ShowHungerIconsSystem.cs]: This null suppression is unnecessary if you is notnullwhen.

**Pull request merged**